lib/readline.js

```diff
@@ -57,7 +57,8 @@ let StringDecoder;
 const kHistorySize = 30;
 const kMincrlfDelay = 100;
 // \r\n, \n, or \r followed by something other than \n
-const lineEnding = /\r?\n|\r(?!\n)/;
+const lineEndingWithSeparator = /(?<=\r?\n|\r(?!\n))/;
+const lineEndingSeparatorInclusion = /[\r?\n|\r(?!\n)]$/;
```

Comment:
The regular expression seems to be faulty. Wrapping the values in the square brackets means that each character is checked for and the special characters loose their meaning and will be checked for as well.

Reply:
thanks for the insight, I'll check and update